src/replacement_files.rs

```diff
@@ -75,28 +75,111 @@ macro_rules! get_from_file_info_indice_index {
 }
 
 impl ModFiles {
+    pub fn reinitialize(&mut self) {
+        let arc = LoadedTables::get_arc_mut();
+        let datas = arc.get_file_datas_mut();
+        for (idx, size) in self.1.iter() {
+            datas[*idx].decomp_size = *size;
+        }
+        let config = CONFIG.read();
+
+        let mut modfiles: HashMap<Hash40, ModFile> = HashMap::new();
+        let mut rejected = Vec::new();
+        let mut stream = HashMap::new();
+
+        // ARC mods
+        if config.paths.arc.exists() {
+            crate::fs::visit::discovery(arc, &config.paths.arc, &mut modfiles, &mut rejected, &mut stream);
+        }
+        // UMM mods
+        if config.paths.umm.exists() {
+            crate::fs::visit::umm_discovery(arc, &config.paths.umm, &mut modfiles, &mut rejected, &mut stream);
+        }
+
+        if let Some(extra_paths) = &config.paths.extra_paths {
+            for path in extra_paths {
```

Comment:
This part might make using `DiscoveryResults` annoying but you can probably just do something like...

```rust
let mut discovered = ...;
for ... {
    discovered = discovered.merge(umm_discovery(...));
}
```

(alternatively it might be a not-bad idea to still pass by reference, but still group all the HashMaps into a single struct)